app/views/rule/actions/_action.html.erb

```diff
@@ -14,8 +15,18 @@
 
     <%= tag.div class: class_names("min-w-1/2 flex items-center gap-2", "hidden" => !needs_value),
                 data: { rule__actions_target: "actionValue" } do %>
-      <span class="font-medium uppercase text-xs">to</span>
-      <%= form.select :value, action.options || [], {}, disabled: !needs_value %>
+      <span class="font-medium uppercase text-xs" data-rule--actions-target="toSpan">to</span>
+      <%# Initial rendering based on rule.action_executors.first from the form. %>
+      <%# This is currently always SetTransactionCategory from app/models/rule/action_executor/transaction_resource.rb. %>
+      <%# Only the first select option will render here, as subsequent renders are injected by the Stimulus controller, which uses the templates from below. %>
+      <% case action_type
```

Comment:
Now that we have the templates and the injection of those templates from the stimulus controller, this `case` becomes a bit weird since it is only ever used for the initial render. The action type for that initial render is determined by the form, which currently has a hardcoded action (and I don't see this changing since we want a first item in the list):

https://github.com/maybe-finance/maybe/blob/441f436187231ae4a0a696d353dcb38bbbf1eed0/app/views/rules/_form.html.erb#L24

I left a comment here explaining this since there is technically unused code here, and if more types come into play they _could_ be put here, but thought I'd get your opinion on what you'd like to see for the initial load vs subsequent injections.

Reply:
For sake of cleaner code without dead paths, I opted to remove the `case` here since that's only required for the initial render. [This commit](https://github.com/maybe-finance/maybe/pull/2175/commits/690daceba20b853fc26186048fcaf81f86f5719f) is the diff between the `case` and no `case`. This `case` isn't actually needed any more because of the template injection. 

I'm happy to revert this if it makes more sense to have it, but I personally like the cleaner code with comment.

Reply:
Another piece of cleanup we can do now that we have template injection on the entire action target (which also means we don't need the `case`), is we no longer have a need for `action_type` or `needs_value`:
- `action_type` was used to determine which one to show for each action. Now that we are injecting rules, this wasn't used. Since the initial load is now hardcoded, we don't really need this
- `needs_value` was used to show/hide the entire action target for cases like `function`. This is also not needed anymore since the controller hides the entire action value for cases that don't need any input: https://github.com/maybe-finance/maybe/pull/2175/files#diff-cd0bc09d066aff1b98283823f34902bd4d57c03b6fd45a3bb2f08ea8f6fa0dadR36-R37